framework/test/jest/unit/specs/modules/chain/dpos/apply.spec.js

```diff
@@ -0,0 +1,466 @@
+
+			// Act
+			await dpos.apply(lastBlockOfTheRound, stubs.tx);
+
+			// Assert
+			expect.assertions(uniqueDelegatesWhoForged);
+			expect(stubs.storage.entities.Account.update).toHaveBeenCalledWith(
+				{
+					publicKey_eq: delegateWhoForgedLast.publicKey,
+				},
+				expect.objectContaining({
+					/**
+					 * Delegate who forged last also forged 3 times,
+					 * Thus will get fee 3 times too.
+					 */
+					fees: delegateWhoForgedLast.fees.add(
+						feePerDelegate * 3 + remainingFee,
+					),
+				}),
+				stubs.tx,
+			);
+
+			uniqueDelegatesWhoForged
+				.filter(d => d.publicKey !== delegateWhoForgedLast.publicKey)
+				.forEach(account => {
+					const blockCount = delegatesWhoForged.filter(
+						d => d.publicKey === account.publicKey,
+					).length;
+					expect(stubs.storage.entities.Account.update).toHaveBeenCalledWith(
+						{
+							publicKey_eq: account.publicKey,
+						},
+						expect.objectContaining({
+							/**
+							 * Rest of the delegates don't get the remaining fee
+							 */
+							fees: account.fees.add(feePerDelegate * blockCount),
+						}),
+						stubs.tx,
+					);
+				});
+		});
+
+		it('should update vote weight of accounts that delegates who forged voted for', async () => {
+			// Act
+			await dpos.apply(lastBlockOfTheRound, stubs.tx);
+
+			// Assert
+			expect.assertions(uniqueDelegatesWhoForged.length);
+			uniqueDelegatesWhoForged.forEach(account => {
+				const { fee, reward } = getTotalEarningsOfDelegate(account);
+				const amount = fee + reward;
+
+				expect(
+					stubs.storage.entities.Account.increaseFieldBy,
+				).toHaveBeenCalledWith(
+					{
+						publicKey_in: account.votedDelegatesPublicKeys,
+					},
+					'voteWeight',
+					amount.toString(),
+					stubs.tx,
+				);
+			});
+		});
+
+		describe('When all delegates successfully forges a block', () => {
+			it('should NOT update "missedBlocks" for anyone', async () => {
+				// Arrange
+				stubs.storage.entities.Round.summedRound.mockResolvedValue([
+					{
+						fees: totalFee,
+						rewards: delegateAccounts.map(() => randomInt(1, 20)),
+						delegates: delegateAccounts.map(a => a.publicKey),
+					},
+				]);
+
+				when(stubs.storage.entities.Account.get)
+					.calledWith(
+						{
+							publicKey_in: delegateAccounts.map(({ publicKey }) => publicKey),
+						},
+						{},
+						stubs.tx,
+					)
+					.mockResolvedValue(delegateAccounts);
+
+				// Act
+				await dpos.apply(lastBlockOfTheRound, stubs.tx);
+
+				expect(
+					stubs.storage.entities.Account.increaseFieldBy,
+				).not.toHaveBeenCalledWith(expect.any, 'missedBlocks');
+			});
+		});
+
+		describe('When summarizing round fails', () => {
+			it('should throw the error message coming from summedRound method', async () => {
+				// Arrange
+				const err = new Error('dummyError');
+				stubs.storage.entities.Round.summedRound.mockRejectedValue(err);
+
+				// Act && Assert
+				await expect(dpos.apply(lastBlockOfTheRound, stubs.tx)).rejects.toBe(
```

Comment:
Shouldn't we check all the assertions which check that missedBlock, productivity etc. are not updated?
Based on the new unit test proposal, we treat module as black box, then how could we could we have ascertained that none of the other steps were performed due to the fail of summedRound function?